### ISSUE TITLE
Module import itself with python3 / absolute_import

### ISSUE THREAD

**Issue by maage** [PyCQA/pylint#2831]
### Steps to reproduce
1. file a/sys.py
```python
"""A."""
from __future__ import absolute_import
from sys import path
print(path)
```
2. Run pylint
```shell
pylint --rcfile=/dev/null a/sys.py 
```

### Current behavior

Current pylint output is:
```
************* Module sys
a/sys.py:3:0: W0406: Module import itself (import-self)

------------------------------------------------------------------
Your code has been rated at 6.67/10 (previous run: 6.67/10, +0.00)

```

### Expected behavior

Pylint should recognize and adhere same module loading as python3 and see sys is python3 sys.

Like this
```sh
% python3 ./a/sys.py 
['/home/user/a', '/usr/lib64/python37.zip', '/usr/lib64/python3.7', '/usr/lib64/python3.7/lib-dynload', '/usr/lib64/python3.7/site-packages', '/usr/lib/python3.7/site-packages']
```

Pylint output should be:
```
------------------------------------
Your code has been rated at 10.00/10

```

### pylint --version output

```
pylint 2.3.1
astroid 2.2.5
Python 3.7.2 (default, Jan 16 2019, 19:49:22) 
[GCC 8.2.1 20181215 (Red Hat 8.2.1-6)]
```

My distro is Fedora 29.

As far as I can see there is no pre package for pylint. And I'm using upstream pylint not pylint from fedora.

**Comment by PCManticore** [comment]
Thank you, I can reproduce the bug.

**Comment by DanielNoord** [comment]
@Pierre-Sassoulas Can't reproduce. Might have been fixed!

**Comment by Pierre-Sassoulas** [comment]
Thank you, for all the work on triaging issues, I appreciate that @DanielNoord
